pkg/homedir/homedir_unix.go

```diff
@@ -133,6 +135,33 @@ func GetConfigHome() (string, error) {
 	return rootlessConfigHomeDir, rootlessConfigHomeDirError
 }
 
+// checkBootId checks if the boot_id of the system has changed since the last time the runtime directory was created.
+func checkBootId(path string) error {
+	systemBootId, err := os.ReadFile("/proc/sys/kernel/random/boot_id")
```

Comment:
This is not supported on all Unixes, at least macOS does not have `/proc`. I didn’t check FreeBSD.

Reply:
when the file doesn't exist, the check is not performed (line 142)

Reply:
Good point, but, still, I think it would be better to avoid that; it leaves noise in syscall traces, and inattentive readers might not notice that the protection is not working on those systems.